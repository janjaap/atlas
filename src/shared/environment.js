export const ENVIRONMENT = {
  DEVELOPMENT: 'DEVELOPMENT',
  ACCEPTATION: 'ACCEPTATION',
  PRE_PRODUCTION: 'PRE_PRODUCTION',
  PRODUCTION: 'PRODUCTION'
};

export const HOSTS = {
  PRODUCTION: 'data.amsterdam.nl',
  PRE_PRODUCTION: 'pre.data.amsterdam.nl',
  ACCEPTATION: 'acc.data.amsterdam.nl',
  DEVELOPMENT: 'localhost'
};

export const getEnvironment = (host) => {
  if (host === HOSTS.PRE_PRODUCTION) {
    return ENVIRONMENT.PRE_PRODUCTION;
  }

  switch (process.env.NODE_ENV) {
    case 'production':
      return ENVIRONMENT.PRODUCTION;
    case 'acceptance':
      return ENVIRONMENT.ACCEPTATION;
    default:
      return ENVIRONMENT.DEVELOPMENT;
  }
};
