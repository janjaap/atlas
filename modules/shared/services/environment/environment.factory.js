import { getEnvironment, ENVIRONMENTS } from '../../../../src/shared/environment';

(function () {
    'use strict';

    angular
        .module('dpShared')
        .factory('environment', environmentFactory);

    environmentFactory.$inject = ['$location'];

    function environmentFactory ($location) {
        const config = {
            NAME: getEnvironment($location.host())
        };
        config.isDevelopment = () => config.NAME === ENVIRONMENTS.DEVELOPMENT;
        return config;
    }
})();
