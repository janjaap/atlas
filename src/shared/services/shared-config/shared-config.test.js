import getSharedConfig from './shared-config';
import { getEnvironment } from '../../environment';

jest.mock('../../environment');

describe('The sharedConfig service', () => {
  it('gives you the configuration based on global environment', () => {
    getEnvironment.mockImplementation(() => ('PRODUCTION'));
    expect(getSharedConfig()).toMatchSnapshot();

    getEnvironment.mockImplementation(() => ('PRE_PRODUCTION'));
    expect(getSharedConfig()).toMatchSnapshot();

    getEnvironment.mockImplementation(() => ('ACCEPTATION'));
    expect(getSharedConfig()).toMatchSnapshot();

    getEnvironment.mockImplementation(() => ('DEVELOPMENT'));
    expect(getSharedConfig()).toMatchSnapshot();
  });

  it('gives you the configuration based on provided environment', () => {
    expect(getSharedConfig({ NAME: 'PRODUCTION' })).toMatchSnapshot();
    expect(getSharedConfig({ NAME: 'PRE_PRODUCTION' })).toMatchSnapshot();
    expect(getSharedConfig({ NAME: 'ACCEPTATION' })).toMatchSnapshot();
    expect(getSharedConfig({ NAME: 'DEVELOPMENT' })).toMatchSnapshot();
  });
});
