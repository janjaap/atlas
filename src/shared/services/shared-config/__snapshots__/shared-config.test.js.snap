// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The sharedConfig service gives you the configuration based on global environment 1`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "STRAATBEELD_THUMB_URL": "panorama/thumbnail/",
  "THUMBNAIL_WIDTH": 240,
}
`;

exports[`The sharedConfig service gives you the configuration based on global environment 2`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "STRAATBEELD_THUMB_URL": "panorama/thumbnail/",
  "THUMBNAIL_WIDTH": 240,
}
`;

exports[`The sharedConfig service gives you the configuration based on global environment 3`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "STRAATBEELD_THUMB_URL": "panorama/thumbnail/",
  "THUMBNAIL_WIDTH": 240,
}
`;

exports[`The sharedConfig service gives you the configuration based on global environment 4`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "STRAATBEELD_THUMB_URL": "panorama/thumbnail/",
  "THUMBNAIL_WIDTH": 240,
}
`;
